Pulled out the SCOPE.md into separate PR (#6)
git-bom/spec: SCOPE.md

# Scope

> Specifies procedures for constructing and conveying artifact trees, directed acyclic graphs (DAGs), and other related 

Reviewer: Suggestion: drop "artifact tree" throughout, since we're using DAG generally in the spec.

Author: I've dropped artifact trees.

Thoughts on 'artifact dependency graphs (ADGs)' as a moniker?  I'm open to considering 'artifact dependency directed acyclic graphs (ADDAGs or 'artifact dependency DAGs).  I'm mostly interested in thoughts and opinions :)

> Specifies procedures for constructing and conveying artifact trees, directed acyclic graphs (DAGs), and other related 

Author: I suspect we don't want to call out from our scope (which is hopefully relatively stable) to the website (where we may have more change).

data structures for artifacts.  Including but not limited to:

> - formats for artifact identifiers

Reviewer 2: ```suggestion
- formats for specifying software artifact identity
```

Author: Fixed in latest commit.

> - formats for artifact identifiers

Author: I suspect we don't want to call out from our scope (which is hopefully relatively stable) to the website (where we may have more change).

> - formats for artifact identifiers

Reviewer 3: Right, we probably want to 
1. define terms in the spec rather than link to terms outside the spec, and 
1. only define them one time. 

So the link for defined terms should be to the [Spec Glossary](https://github.com/git-bom/spec/blob/main/SPEC.md#terms-and-definitions-mandatory). I believe the link above meets these requirements.

> - formats for artifact identifiers

Reviewer 3: Yes, works for me.

> - formats for specifying graph relationships between artifacts

Reviewer: What's a "relationship" here?

Reviewer 2: I prefer 'graph' here, as 'parent-child' is unnecessarily specific for the SCOPE document, in my opinion.

Reviewer 3: Another option, which should be quite broad:
```suggestion
- formats for specifying relationships between GitBOM ADG nodes and edges, e.g.,node-node, node-edge, and edge-edge relationships
```
This line describing relationships could also be inclusive of relationships between the ADG and other metadata, or it could remain split from that relationship concept which is currently described on line 10

- manner of embedding identifiers for ADGs, and other related
data structures in artifacts of various types
- guidance on metadata which references ADGs, and other related data structures

> - guidance for build tools for:

Reviewer: Given my suggestion of "build process" above, it's probably worth also talking about the relationship between "build tool" and "build action". Something like the former is a specific configuration of a tool on a set of inputs, and the latter is an invocation of a tool in that configuration.

>   - constructing ADGss, and other related data structures

Reviewer 4: Remove the extra "s"

Author: Fixed in latest commit

>   - constructing ADGss, and other related data structures

Reviewer 2: ```suggestion
  - constructing ADGs, and other related data structures
```

Author: Fixed in latest commit.

  - conveying ADGs, and other related data structures
  - embedding identifiers for ADGs, and other related data structures ids in artifacts
  - manners of conveyance of ADGs, and other related data structures

>   - descriptions of use cases for which directed acyclic graphs, and other related data structures may be used

Reviewer 2: ```suggestion
  - descriptions of use cases for which ADGs, and other related data structures, may be used
```

Author: Fixed in latest commit.

>   - descriptions of use cases for which artifact trees, directed acyclic graphs, and other related data structures may be used

Reviewer: "directed acyclic graph" -> DAG for consistency

>   - guidance on metadata which references artifact trees, DAGs, and other related data structures

Author: I believe I've captured this change in current revisions, please let me know :)
